### PR TITLE
Update documents with new APIs from missing old APIs

### PR DESCRIPTION
**Pull request by axross**
I have done the tutorial. I think this is great project!

I found and fixed the few description with old APIs in document.
Please check and merge to fix it.

### CODE REVIEW

**Comment by joeconwaystk** [comment]
Yes, this looks great. Thank you so much, I'm glad someone caught the errors. 

There is one place in this PR where there needs to be a small change, I'll make a comment on that line in this PR, merge the PR, and then add another commit.

File: docs/tut/model-relationships-and-joins.md [hunk 1]
```diff
@@ -201,15 +201,15 @@ The inverse relationship doesn't have to be updated - whether it is has-one or h
...
 
-Each returned `Question` would also have a `ManagedSet` of `Answer`s in its `answers` property. You may also filter which answers are returned for each `Question` by nesting `matchOn` properties.
+Each returned `Question` would also have a `ManagedSet` of `Answer`s in its `answers` property. You may also filter which answers are returned for each `Question` by nesting `where` properties.
 
 ```dart
 var query = new Query<Question>()
-  ..matchOn.answers.includeInResultSet = true
-  ..matchOn.answers.matchOn.isCorrect = whereEqualTo(true);
+  ..where.answers.includeInResultSet = true
+  ..where.answers.where.isCorrect = whereEqualTo(true);
```
Comment (joeconwaystk) [inline docs/tut/model-relationships-and-joins.md]: this is:

```
..joinMany((question) => question.answers).where.isCorrect = whereEqualTo(true);
```

or

```
var answerJoinQuery = query.joinMany((question) => question.answers)
  ..where.isCorrect = whereEqualTo(true);
```

**Comment by joeconwaystk** [comment]
All deployed with your changes - thanks, that's a really important part of the documentation.

**Comment by axross** [comment]
I feel honored!

File: docs/tut/model-relationships-and-joins.md (hunk 1 above)
Comment (axross) [inline docs/tut/model-relationships-and-joins.md]: Oh, I missed to changed this. I'm sorry...

**Pull request merged**